src/main/java/org/cbioportal/session_service/web/SessionServiceController.java

```diff
@@ -66,46 +72,56 @@
     @RequestMapping(method = RequestMethod.POST, value="/{source}/{type}")
     @JsonView(Session.Views.IdOnly.class)
     public Session addSession(@PathVariable String source, 
-        @PathVariable String type, 
+        @PathVariable SessionType type, 
         @RequestBody String data) { 
         return sessionService.addSession(source, type, data);
     }
 
     @RequestMapping(method = RequestMethod.GET, value="/{source}/{type}")
     @JsonView(Session.Views.Full.class)
     public Iterable<Session> getSessions(@PathVariable String source, 
-        @PathVariable String type) {
+        @PathVariable SessionType type) {
         return sessionService.getSessions(source, type);
     }
-
+    
     @RequestMapping(method = RequestMethod.GET, value="/{source}/{type}/query")
     @JsonView(Session.Views.Full.class)
-    public Iterable<Session> getSessionsByQuery(@PathVariable String source, 
-        @PathVariable String type, 
+    public Iterable<Session> getSessionsBySingleFieldQuery(@PathVariable String source, 
+        @PathVariable SessionType type, 
         @RequestParam(name="field") String field,
         @RequestParam(name="value") String value) {
-        return sessionService.getSessionsByQuery(source, type, field, value);
+    	String query = "{\""+field+"\":\""+value+"\"}";
+    	return sessionService.getSessionsByQuery(source, type, query);
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value="/{source}/{type}/query/fetch")
+    @JsonView(Session.Views.Full.class)
+    public Iterable<Session> fetchSessionsByQuery(@PathVariable String source, 
+        @PathVariable SessionType type,
+        @ApiParam(required = true, value = "selection filter similar to mongo filter")
+        @RequestBody String query) {
+        return sessionService.getSessionsByQuery(source, type, query);
     }
 
     @RequestMapping(value = "/{source}/{type}/{id}", method = RequestMethod.GET)
     @JsonView(Session.Views.Full.class)
     public Session getSession(@PathVariable String source, 
-        @PathVariable String type,
+        @PathVariable SessionType type,
```

Comment:
@kalletlak will this endpoint still work in the same manner? Is SessionType just a normal string in the URL or some integer?

Reply:
ah i see in the tests a normal string works 👍